builtin/sparse-checkout.c

```diff
@@ -343,11 +405,11 @@ static int sparse_checkout_set(int argc, const char **argv, const char *prefix)
 			while (!strbuf_getline(&line, stdin)) {
 				size_t len;
 				char *buf = strbuf_detach(&line, &len);
-				add_pattern(buf, buf, len, &pl, 0);
+				add_pattern(buf, empty_base, 0, &pl, 0);
```

Comment:
Will you explain this change for me?

Reply:
Before, we were not using the `pattern_list` for anything other than tracking the patterns to set in the sparse-checkout file.

Now, we want to use the `pattern_list` in-memory to update the working directory.

The previous way I was calling `add_pattern()` was wrong and led to errors when using these patterns in-memory.